src/PugTemplateRenderer.php

```diff
@@ -44,14 +45,7 @@ public function __construct(Pug $pug, array $defaultParams, array $globals, arra
 
     private function setDefaultParams(array $defaultParams): void
     {
-        foreach ($defaultParams as $name => $params) {
-            if (empty($this->globals[$name])) {
-                $this->globals[$name] = [];
-            }
-            foreach ($params as $param => $value) {
-                $this->globals[$name][$param] = $value;
-            }
-        }
+        $this->globals = array_replace_recursive($this->globals, $defaultParams);
```

Comment:
```suggestion
        $this->globals = array_replace_recursive($this->globals, $defaultParams) ?? [];
```
![image](https://user-images.githubusercontent.com/1093654/58371502-f69a7500-7f11-11e9-8cd5-44185f5b28c4.png)

it seems like a false positive but without this static analisys fails

Reply:
> ![image](https://user-images.githubusercontent.com/1093654/58371502-f69a7500-7f11-11e9-8cd5-44185f5b28c4.png)
> 
> it seems like a false positive but without this static analisys fails

I did a commit with a casting to array but I don't know because it doesn't appear. :S